lib/shanty/plugin_dsl.rb

```diff
@@ -13,5 +17,9 @@ def provides_projects_containing(*globs)
     def provides_tags(*args)
       tags.concat(args.map(&:to_sym))
     end
+
+    def provides_config(key, default = nil)
+      config[key] = default
```

Comment:
You should prevent the default value here being something other than a simple value. I imagine it would be enough to fail if `is_a?(Enumerable) == true`.